system/density/buffers/buffer.c

```diff
@@ -0,0 +1,184 @@
+/*
```

Comment:
The normal solution to the 'incompatibility' dilemma - assuming the third-party library in question is hosted on github, is to fork it to the hpccsystems github account and then reference that fork using git submodule. That way we are insulated from any changes on the upstream repository but can easily take them when we want to.